include/rcpputils/filesystem_helper.hpp

```diff
@@ -352,6 +360,58 @@ RCPPUTILS_PUBLIC bool operator!=(const path & a, const path & b);
 */
 RCPPUTILS_PUBLIC std::ostream & operator<<(std::ostream & os, const path & p);
 
+// remove warning suppression
+#if !defined(_WIN32)
+# pragma GCC diagnostic pop
+#else  // !defined(_WIN32)
+# pragma warning(pop)
+#endif
+
+/**
+ * \brief Get a path to a location in the temporary directory, if it's available.
+ *
+ * This does not create any directories.
+ * On Windows, this uses "GetTempPathA"
+ * On non-Windows, this prefers the environment variable TMPDIR, falling back to /tmp
+ *
+ * \return A path to a directory for storing temporary files and directories.
+ */
+RCPPUTILS_PUBLIC std::filesystem::path temporal_directory_path();
```

Comment:
While "temporal" isn't exactly wrong, it is also not a common name for this.  The word "temporary" is much more common.  So I'm going to suggest that we change it:

```suggestion
RCPPUTILS_PUBLIC std::filesystem::path temporary_directory_path();
```